healthcheck.go

```diff
@@ -53,6 +55,73 @@ type ClusterHealth struct {
 	Status string `json:"status"`
 }
 
+//indexcheck calls elasticsearch to check if an index from the client exists
+//What is a context?
+func (cli *Client) indexcheck(ctx context.Context) (code int, err error) {
+
+	clientElasticSearchIndices := cli.indices
+	/*
+		Is there a way write a test here maybe:
+		clientElasticSearchIndices := cli.indices
+	*/
+	logData := log.Data{"clientIndexResponse": cli.indices}
+	if err != nil {
+		log.Event(ctx, "failed to retrieve the indices from the client", log.ERROR, logData, log.Error(ErrorRetrievingClientIndices))
+		return 500, ErrorRetrievingClientIndices
+	}
+
+	for _, index := range clientElasticSearchIndices {
+
+		urlIndex := cli.url + "/" + index
+		logData := log.Data{"url": urlIndex}
```

Comment:
Since we are inside a loop here, it would likely be useful to add the `index` as a field to the `logData` map in order to provide better context for any logging/errors that arise. Although this information is used in the url, calling it out explicitly provides clearer context of where the execution has got to.
```suggestion
		logData := log.Data{"url": urlIndex, "index": index}
```